File: spring/src/main/java/com/moseoh/assistant/controller/UserController.java
package com.moseoh.assistant.controller;

import com.moseoh.assistant.response.Response;
import com.moseoh.assistant.service.UserService;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiImplicitParam;
import io.swagger.annotations.ApiImplicitParams;
import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiParam;
import lombok.RequiredArgsConstructor;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@Api(tags = { "1. User" })
@RequiredArgsConstructor
@RestController
@RequestMapping("/v1/user")
@ApiImplicitParams({
                @ApiImplicitParam(name = "X-AUTH-TOKEN", value = "AccessToken", required = true, dataType = "String", paramType = "header")
})
public class UserController {

        private final UserService userService;

        @ApiOperation(value = "모든 회원 조회", notes = "모든 회원 목록을 조회합니다.")
        @GetMapping("/list")
        public ResponseEntity<Response> getUsers() {
                return Response.toResponseEntity(userService.getUserList());
        }

        @ApiOperation(value = "회원 조회", notes = "회원을 조회합니다.")
        @GetMapping("/{id}")
        public ResponseEntity<Response> getUser(
                        @ApiParam(value = "회원 pk", required = true) @PathVariable long id) {
                return Response.toResponseEntity(userService.getUserDto(id));
        }

}

File: spring/src/main/kotlin/com/server/assistant/controller/UserController.kt
package com.server.assistant.controller

import com.server.assistant.dto.CreateUserDTO
import com.server.assistant.dto.LoginUserDTO
import com.server.assistant.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

/**
 * UserController
 * 주소
 * GitHub : http://github.com/azqazq195
 * Created by azqazq195@gmail.com on 2021-10-31
 */
@RestController
class UserController {

    @Autowired
    private lateinit var userService: UserService

    @GetMapping("/users", produces = ["application/json"])
    fun getUsers(): ResponseEntity<Any> {
        return ResponseEntity.ok().body(userService.getUsers())
    }

    @GetMapping("/user/{id}", produces = ["application/json"])
    fun getUser(@PathVariable id: Long): ResponseEntity<Any> {
        return ResponseEntity.ok().body(userService.getUser(id))
    }

    @PostMapping("/user")
    fun createUser(@RequestBody createUserDTO: CreateUserDTO): ResponseEntity<Any> {
        return ResponseEntity.ok().body(userService.createUser(createUserDTO))
    }

    @PostMapping("/login", produces = ["application/json"])
    fun login(@RequestBody loginUserDTO: LoginUserDTO): ResponseEntity<Any> {
        return ResponseEntity.ok().body(userService.login(loginUserDTO))
    }
}